giskard/scanner/scanner.py

```diff
@@ -213,16 +223,33 @@ def _validate_model_and_dataset(self, model, dataset):
                 column_types=dataset.column_types,
             )
 
-        num_features = len(model.meta.feature_names) if model.meta.feature_names else len(dataset.columns)
-        if num_features > 100:
+        return model, dataset, model_validation_time
+
+    def _validate_features(
+        self, features: Optional[Sequence[str]], model: BaseModel, dataset: Optional[Dataset] = None
+    ):
+        _default_features = model.meta.feature_names or dataset.columns.drop(dataset.target, errors="ignore")
+
+        if features is None:
+            features = _default_features
+        else:
+            if not set(features).issubset(_default_features):
+                raise ValueError(
+                    "The `features` argument contains invalid feature names: "
+                    f"{', '.join(set(features) - set(_default_features))}. "
+                    f"Valid features for this model are: {', '.join(_default_features)}."
+                )
+
+        if len(features) > 100:
```

Comment:
We should probably also validate empty features since some scanner fail due to / by 0
![image](https://github.com/Giskard-AI/giskard/assets/114553769/1408496e-68d2-4d0e-8649-6dd742f1b858)